Tests/DatadogIntegrationTests/Scenarios/RUM/RUMManualInstrumentationScenarioTests.swift

```diff
@@ -55,7 +55,7 @@ class RUMManualInstrumentationScenarioTests: IntegrationTests, RUMCommonAsserts
         let view1 = session.viewVisits[0]
         XCTAssertEqual(view1.name, "SendRUMFixture1View")
         XCTAssertEqual(view1.path, "Example.SendRUMFixture1ViewController")
-        XCTAssertEqual(view1.viewEvents.count, 6, "First view should receive 6 updates")
+        XCTAssertEqual(view1.viewEvents.count, 7, "First view should receive 7 updates")
```

Comment:
Good catch 👀 !

There is a difference actually. In that scenario we have an active `RUMUserActionScope`, when the last `RUMStopViewCommand` comes in, it is passed and used by the `RUMUserActionScope` and then by `self`/`RUMViewScope`.

Previously this would all be synchronous and done by the end of the `RUMViewScope.process` method, the `RUMUserActionScope` will be closed and so it would lead to a single `RUMViewEvent` with an updated action count.

Now the `RUMUserActionScope` will send an `RUMActionEvent` and wait to know if it was discarded or not before closing. When it eventually closes, the `RUMViewScope` will have to send a new `RUMViewEvent` with an increased `action.count` (or do nothing if it was dropped), and this is the additional event.

Reply:
I see 👌 , some change at the tail of the events stream due to asynchronicity we introduce 👍.